Added new kb article dropdownbutton-add-separator-between-items (#2834)
telerik/blazor-docs: knowledge-base/dropdownbutton-add-separator-between-items.md

---
title: Adding a Separator Between Items in a DropDownButton for Blazor
description: Learn how to add a separator between the items in a DropDownButton by utilizing custom styling in a Blazor application.
type: how-to
page_title: How to Implement Item Separators in a Telerik Blazor DropDownButton
slug: dropdownbutton-kb-add-separator-between-items
tags: dropdownbutton, blazor, item separator, custom styling
res_type: kb
ticketid: 1680307
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>DropDownButton for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I want to visually separate the items in a [DropDownButton](slug:dropdownbutton-overview) to enhance the UI and improve user experience. How to add a separator (divider) between the items?

## Solution

> The DropDownButton does not have a concept for a "separator" item (similar to the [ContextMenu](slug:contextmenu-data-binding-overview), for example). However, the DropDownButton uses a declarative approach for its items, so you can add any desired item.

Reviewer: Start with a positive statement - how to achieve the desired result. If it's important to confirm something negative, you can add that later.

Reviewer 2: ```suggestion
The DropDownButton uses a declarative approach for its items, so you can add any desired item. This component does not have a concept for a "separator" item (similar to the [ContextMenu](slug:contextmenu-data-binding-overview), for example) but you can use a custom approach to create a separator.
```

> The specific is that the content you declare in the DropDownButtonItem will be rendered inside the `<span class="k-menu-link-text">` which is narrower and thus your custom separator will not cover the whole width of the popup. This article demonstrates how to add a separator by utilizing custom styling.

Reviewer: This paragraph confused me at first reading. The first and second sentence must be reworded, so that it's clear that the second sentence presents an alternative to the problem in the first one.

"Use" is simpler than "utilize".

```suggestion
The content of the DropDownButtonItem is rendered inside a `<span class="k-menu-link-text">` element, which is narrower. Thus, a border inside the dropdown item content cannot span over the whole width of the popup. This article demonstrates how to add a separator by setting a CSS class on the whole item and using custom styling.
```

To create a visual separator between items in a DropDownButton, apply a custom class to the item that precedes the separator and style its bottom border. This approach allows you to simulate a separator without the need for a dedicated separator item. The custom class can be applied directly in the item declarations in your Blazor component.

>caption Add separator between DropDownButton items

````RAZOR
<style>
    .item-with-separator {
        border-bottom: 1px solid #e5e5e5;
    }
 </style>

> <TelerikDropDownButton Icon="@SvgIcon.Share" OnClick="@(()=>OnItemClick("Primary"))">

Reviewer: ```suggestion
<TelerikDropDownButton Icon="@SvgIcon.Share">
```

    <DropDownButtonContent>Share</DropDownButtonContent>

    <DropDownButtonItems>

>         <DropDownButtonItem Icon="@SvgIcon.Facebook" OnClick="@(()=>OnItemClick("Facebook"))">Facebook</DropDownButtonItem>
>         <DropDownButtonItem Icon="@SvgIcon.Twitter" OnClick="@(()=>OnItemClick("Twitter"))">Twitter</DropDownButtonItem>
>         <DropDownButtonItem Icon="@SvgIcon.Linkedin" OnClick="@(()=>OnItemClick("Linkedin"))" Class="item-with-separator">Linkedin</DropDownButtonItem>
>        <DropDownButtonItem OnClick="@(()=>OnItemClick("Reddit"))">Other</DropDownButtonItem>

Reviewer: Make the important code more visible and simplify the example.

```suggestion
        <DropDownButtonItem Icon="@SvgIcon.Facebook">Facebook</DropDownButtonItem>
        <DropDownButtonItem Icon="@SvgIcon.Twitter">Twitter</DropDownButtonItem>
        <DropDownButtonItem Icon="@SvgIcon.Linkedin" Class="item-with-separator">Linkedin</DropDownButtonItem>
       <DropDownButtonItem>Other</DropDownButtonItem>
```

    </DropDownButtonItems>

</TelerikDropDownButton>

> 
> @code {
>     private void OnItemClick(string item)
>     {
>         Console.WriteLine($"User clicked {item} option.");
>     }
> }

Reviewer: ```suggestion
```
